tests/pass-dep/libc/gettid.rs

```diff
@@ -0,0 +1,15 @@
+//@only-target-linux
```

Comment:
You added this function for all Unix hosts, so just testing it on Linux is not sufficient.

`ignore-target-windows` is probably better.

Reply:
It seems that macos doesn't have a `gettid` function in its libc. In the libc crate there is a `gettid` function in `unix/linux-like` and `unix/nto`. Would it make more sense to only support `gettid` for linux targets instead?

Reply:
Yes in that case please move the gettid shim to `linux/foreign_items.rs` and also add an assertion inside `fn gettid` to ensure that it is only used on Linux.

Reply:
done.